src/main/java/gregtech/integration/jei/recipe/primitive/MaterialTreeCategory.java

```diff
@@ -0,0 +1,260 @@
+        drawArrow(minecraft, "right_7", 22, 107, !itemExists.get(3) &&
+                itemExists.get(0) && (itemExists.get(4) || itemExists.get(5)));
+        // foil -> fine wire
+        drawArrow(minecraft, "right_7", 147, 153, itemExists.get(24) && itemExists.get(26));
+        // rod -> ring
+        drawArrow(minecraft, "right_57", 72, 60, itemExists.get(7) && itemExists.get(14));
+        // plate -> foil/lens
+        drawArrow(minecraft, "right_57", 72, 155, itemExists.get(19) &&
+                (itemExists.get(24) || itemExists.get(25)));
+        // small dust <-> dust
+        drawArrow(minecraft, "dh_down_16", 10, 85, itemExists.get(1) && itemExists.get(0));
+        // tiny dust <-> dust
+        drawArrow(minecraft, "dh_down_16", 10, 119, itemExists.get(2) && itemExists.get(0));
+        // block <-> ingot/gem
+        drawArrow(minecraft, "dh_down_16", 35, 119, itemExists.get(6) &&
+                (itemExists.get(4) || itemExists.get(5)));
+        // block -> plate
+        drawArrow(minecraft, "down_right_8_17", 37, 153, itemExists.get(6) && itemExists.get(19));
+        // long rod -> rod
+        drawArrow(minecraft, "left_up_16_5", 61, 73, itemExists.get(8) && itemExists.get(7));
+        // plate -> dense plate
+        drawArrow(minecraft, "right_up_16_5", 72, 153, itemExists.get(19) && itemExists.get(20));
+        // plate -> small gear
+        drawArrow(minecraft, "right_up_41_5", 72, 153, itemExists.get(19) && itemExists.get(22));
+        // rod -> long rod
+        drawArrow(minecraft, "right_down_16_5", 72, 62, itemExists.get(7) && itemExists.get(8));
+        // rod -> bolt
+        drawArrow(minecraft, "right_down_41_5", 72, 62, itemExists.get(7) && itemExists.get(11));
+        // dust <-> block
+        // only if no ingot or gem
+        drawArrow(minecraft, "dh_right_down_16_26", 22, 107, !itemExists.get(4) &&
+                !itemExists.get(5) && itemExists.get(0) && itemExists.get(6));
+        // dust -> hot ingot
+        drawArrow(minecraft, "right_up_right_3_35_4", 22, 73, itemExists.get(0) && itemExists.get(3));
+        // ingot -> plate
+        drawArrow(minecraft, "right_down_right_3_43_4", 47, 109, itemExists.get(4) && itemExists.get(19));
+        // ingot -> nugget
+        drawArrow(minecraft, "right_up_right_3_17_4", 47, 91, itemExists.get(4) && itemExists.get(17));
+        // rod -> gear
+        drawArrow(minecraft, "right_down_right_53_31_4", 72, 62, itemExists.get(7) && itemExists.get(15));
+        // ingot/gem -> rod
+        drawArrow(minecraft, "right_up_right_3_47_4", 47, 61, itemExists.get(7) &&
+                (itemExists.get(4) || itemExists.get(5)));
+        // rod -> frame box
+        drawArrow(minecraft, "right_down_right_53_64_4", 72, 62, itemExists.get(7) && itemExists.get(16));
+
+        // material info rendering
+		minecraft.fontRenderer.drawString(materialName, 0, 0, 0x111111);
+		if (minecraft.fontRenderer.getStringWidth(materialFormula) > 176) {
```

Comment:
I think the text entries should all shift up by one (or two) lines if a material is missing a Formula or EBF temperature. The spaces in the middle break up the page too much, in my opinion.

Reply:
makes sense, I was considering doing that after the previous commit but decided to wait for more feedback